elk/train.py

```diff
@@ -1,79 +1,148 @@
+
+    pbar = tqdm(zip(train_layers, val_layers), total=L, unit="layer")
+    writer = csv.writer(open(cache_dir / "eval.csv", "w"))
+    writer.writerow(
+        [
+            "layer",
+            "train_loss",
+            "loss",
+            "acc",
+            "cal_acc",
+            "auroc",
+            "lr_auroc",
+            "lr_acc",
+        ]
     )
-    print("Done.")
 
-    return logistic_regression_model, ccs_model
+    for train_h, val_h in pbar:
+        # TODO: Once we implement cross-validation for CCS, we should benchmark against
+        # LogisticRegressionCV here.
+        pbar.set_description("Fitting LR")
+        lr_model = LogisticRegression(max_iter=10000, n_jobs=1, C=0.1)
+        lr_model.fit(train_h, train_labels)
+
+        lr_preds = lr_model.predict_proba(val_h)[:, 1]
+        lr_acc = accuracy_score(val_labels, lr_preds > 0.5)
+        lr_auroc = roc_auc_score(val_labels, lr_preds)
+
+        pbar.set_description("Fitting CCS")
+        x0, x1 = train_h.to(args.device).chunk(2, dim=-1)
+        val_x0, val_x1 = val_h.to(args.device).chunk(2, dim=-1)
+
+        ccs_model = CCS(in_features=x0.shape[-1], device=args.device, loss=args.loss)
+        train_loss = ccs_model.fit(
+            data=(x0, x1),
+            optimizer=args.optimizer,
+            verbose=False,
+            weight_decay=args.weight_decay,
+        )
+        val_result = ccs_model.score(
+            (val_x0, val_x1),
+            torch.tensor(val_labels, device=args.device),
+        )
+        pbar.set_postfix(ccs_auroc=val_result.auroc, lr_auroc=lr_auroc)
+        stats = [train_loss, *val_result, lr_auroc, lr_acc]
+        writer.writerow([L - pbar.n] + [f"{s:.4f}" for s in stats])
+
+        lr_models.append(lr_model)
+        ccs_models.append(ccs_model)
+
+    ccs_models.reverse()
+    lr_models.reverse()
+    return lr_models, ccs_models
 
 
 if __name__ == "__main__":
-    args = get_args(default_config_path=Path(__file__).parent / "default_config.json")
-    print(f"-------- args = {args} --------")
+    parser = ArgumentParser()
+    parser.add_argument("name", type=str, help="Name of the experiment")
+    parser.add_argument(
+        "--device",
+        type=str,
+        help="PyTorch device to use. Default is cuda:0 if available.",
+    )
+    parser.add_argument(
+        "--loss",
+        type=str,
+        default="squared",
+        choices=("js", "squared"),
+        help="Loss function used for CCS.",
+    )
+    parser.add_argument(
+        "--optimizer",
+        type=str,
+        default="lbfgs",
+        choices=("adam", "lbfgs"),
+        help="Optimizer for CCS. Should be adam or lbfgs.",
+    )
+    parser.add_argument("--seed", type=int, default=0)
+    parser.add_argument(
+        "--weight-decay",
+        type=float,
+        default=0.01,
+        help=(
+            "Weight decay for CCS when using Adam. Used as L2 regularization in LBFGS."
+        ),
+    )
+    args = parser.parse_args()
```

Comment:
I mean this is wrong? No model name, no dataset, ... The train.sh script doesn't work.

Reply:
`train.sh` now no longer exists